tracee-ebpf/tracee/net_events.go

```diff
@@ -206,3 +176,48 @@ func parseEventMetaData(payloadBytes []byte) (EventMeta, *bytes.Buffer) {
 	return eventMetaData, bytes.NewBuffer(payloadBytes[32:])
 
 }
+
+func netPacketProtocolHandler(buffer *bytes.Buffer, evtMeta EventMeta, ctx procinfo.ProcessCtx, eventName string, bootTime uint64) (external.Event, PacketMeta) {
+	var evt external.Event
+	packet, err := ParseNetPacketMetaData(buffer)
+	if err != nil {
+		return evt, packet
+	}
+	evt = CreateNetEvent(evtMeta, ctx)
+	CreateNetPacketMetaArgs(&evt, packet)
+	evt.EventName = eventName
+	evt.Timestamp = int(evtMeta.TimeStamp + bootTime)
+	return evt, packet
+}
+
+// parsing the PacketMeta struct from bytes.buffer
+func ParseNetPacketMetaData(payload *bytes.Buffer) (PacketMeta, error) {
+	var pktMetaData PacketMeta
+	err := binary.Read(payload, binary.LittleEndian, &pktMetaData)
```

Comment:
I would add `PackMeta` as part of `bufferdecoder` protocol, and add the relevant decoding function (with unit tests and benchamarks). WDYT?

Reply:
TBH IDK , because in the following PR we will created package that handles the all network events and also adding new events (such detection for http , irc,  dns and icmp ) so for every protocol will have a specific parser (that  parse form byte array to the asked protocol struct).
WDYT?